checkers/invalid_range_index_checker.py

```diff
@@ -0,0 +1,50 @@
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from ast import literal_eval
+
+
+class InvalidRangeIndexChecker(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = 'custom'
+    msgs = {'E9993':
+                ('You should not use invalid range index on line %s',
+                 'invalid-range-index',
+                 'Used when you use invalid index range')}
+    options = ()
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @check_messages("invalid-range-index")
+    def visit_call(self, node):
+        if isinstance(node.func, astroid.Name):
+            name = node.func.name
+            # ignore the name if it's not a builtin (i.e. not defined in the
+            # locals nor globals scope)
+            if not (name in node.frame() or name in node.root()) and name == 'range':
+                # the arguments of 'range' call
+                arg = node.args
```

Comment:
Please use the `map` function to make the calls to `as_string` and `literal_eval` here, rather than separately later.